main.cpp

```diff
@@ -216,293 +246,292 @@ struct Computer
-    // color (std::string)
     std::string color = "gray";
-    // number of languages (int)
     int numberOfLanguages = 9;
-    // default language (std::string)
     std::string language = "english";
 
-    // show Netflix 
     void showNetflix(std::string showTitle, int season, int episode);
-    // show TV-Guide
     void showTVGuide(int week);
-    // activate recording
     void activateRecoding(int channel);
 };
 
-/*
-Thing 9)
-5 properties: manufacturer
-    1) name (float)
-    2) headquarters location (std::string)
-    3) founding year (int)
-    4) name of founder (std::string)
-    5) annual profit (float)
-3 things it can do:
-    1) go bankrupt
-    2) release new model
-    3) buy rival company
- */ 
+void TvOnScreenMenu::showNetflix(std::string showTitle, int season, int episode)
+{
+    showTitle = "Breaking Bad";
+    season = 2;
+    episode = 5;
+}
+
+void TvOnScreenMenu::showTVGuide(int week)
+{
+    week = 22;
+}
+
+void TvOnScreenMenu::activateRecoding(int channel)
+{
+    channel += 1;
+}
+
 
 struct TvManufacturer
 {
-    // name (float)
     std::string name = "SONY";
-    // headquarters location (std::string)
     std::string headquartersLocation = "Japan";
-    // founding year (int)
     int foundingYear = 1946;
-    // name of founder (std::string)
     std::string founder = "Masaru Ibuka";
-    // annual profit (float)
     float annualProfit = 1178.8f;
 
-    // go bankrupt
     void goBankrupt(bool payAllDebts);
-    // release new model
     bool releasNewModel(std::string modelName, float price);
-    // buy rival company
     bool buyRivalCompany(std:: string company, float price);
 };
 
-/*
-Thing 10) tv
-5 properties:
-    1) screen 
-    2) remote control
-    3) connectors
-    4) on screen menu
-    5) manufacturer
-3 things it can do:
-    1) change channel
-    2) change volume
-    3) activate smart tv
- */
+void TvManufacturer::goBankrupt(bool payAllDebts)
+{
+    if (payAllDebts)
+    {
+        payAllDebts = false;
+    } // they will never pay the debts ;-)
+}
+
+bool TvManufacturer::releasNewModel(std::string modelName, float price)
+{
+    modelName = "XYZ123";
+    price += 200;
+    return true;
+}
+
+bool buyRivalCompany(std:: string company, float price)
+{
+    company = "Samsung";
+    price = 1.99f;
+    return false;
+}
+
 
 struct Tv
 {
-    // screen 
     TvScreen screen;
-    // remote control
     TvRemoteControl remoteControl;
-    // connectors
     TvConnectors connectors;
-    // on screen menu
     TvOnScreenMenu onScreenMenu;
-    // manufacturer
     TvManufacturer manufacturer;
 
-    // change channel
     void changeChannel(int channelNumber, bool useRemoteControl);
-    // change volume
     void changeVolume(float newVolume, bool useRemoteControl);
-    // activate smart tv
     void activateSmartTv(int menuItem = 0);
 };
 
+void Tv::changeChannel(int channelNumber, bool useRemoteControl)
+{
+    if (useRemoteControl)
+    {
+        channelNumber++;
+    }
```

Comment:
Everywhere that you have `i++`, `i--`, `i -= 1;` or `i += 1;`, you should use `pre-increment/decrement`. 
You are currently doing `post-increment/decrement`.  
you can think of post- and pre- increment like this:
```
int preIncrement(int& i)
{
    i += 1;
    return i; //both the returned value AND i have changed
}
​
int postIncrement(int& i)
{
    int oldVal = i;
    i += 1;
    return oldVal; //i has changed but the value being returned does not reflect this change.
}
```
more information can be found here explaining why `pre-increment/decrement` is preferred over `post-increment/decrement`: https://en.cppreference.com/w/cpp/language/operator_incdec

Reply:
good to know...